### PR TITLE
{KeyVault} Support more long-running operations

### PR DESCRIPTION
**Pull request by bim-msft**
Support more long-running operations:
1. `az keyvault recover`
2. `az keyvault network-rule add`
3. `az keyvault network-rule remove`
4. `az keyvault set-policy`
5. `az keyvault delete-policy`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
KeyVault

File: src/azure-cli/azure/cli/command_modules/keyvault/custom.py [hunk 1]
```diff
@@ -277,7 +277,7 @@ def get_default_policy(cmd, client, scaffold=False):  # pylint: disable=unused-a
     return _scaffold_certificate_profile(cmd) if scaffold else _default_certificate_profile(cmd)
 
 
-def recover_keyvault(cmd, client, vault_name, resource_group_name, location):
+def recover_keyvault(cmd, client, vault_name, resource_group_name, location, no_wait=False):
```
Comment (yungezz) [inline src/azure-cli/azure/cli/command_modules/keyvault/custom.py:280]: will this be a breaking change to user if no_wait default false?

Comment (bim-msft) [reply src/azure-cli/azure/cli/command_modules/keyvault/custom.py:280]: > will this be a breaking change to user if no_wait default false?

@yungezz No, it's not a breaking change. The previous behavior is synchronous as well.

File: src/azure-cli/azure/cli/command_modules/keyvault/custom.py [hunk 2]
```diff
@@ -666,10 +669,12 @@ def add_network_rule(cmd, client, resource_group_name, vault_name, ip_address=No
                                 parameters=VaultCreateOrUpdateParameters(
                                     location=vault.location,
                                     tags=vault.tags,
-                                    properties=vault.properties))
+                                    properties=vault.properties),
+                                no_wait=no_wait)
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/keyvault/custom.py:673]: indent not aligned

File: src/azure-cli/azure/cli/command_modules/keyvault/custom.py [hunk 3]
```diff
@@ -621,10 +622,12 @@ def set_policy(cmd, client, resource_group_name, vault_name,
...
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/keyvault/custom.py:626]: indent not aligned

**Comment by qwordy** [comment]
Do they all have `wait` commands?

**Comment by bim-msft** [comment]
> Do they all have `wait` commands?

@qwordy Yes, all of these commands have `wait` commands under their command groups. This has been added previously.

**Review by qwordy — APPROVED** [review]

**Comment by yungezz** [comment]
hi @fengzhou-msft could you pls help to review? thanks

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**